CODEWALK.md

```diff
@@ -302,6 +308,58 @@ A null block count indicates the absence of any blocks mined in a previous round
+A client also creates a payment channel so the miner can be paid over time for storing the piece. 
+The miner responds with acceptance or otherwise.
+
+When proposing a deal, a client loads the piece data into its [IPFS block service](https://github.com/ipfs/go-blockservice). 
+This advertises the availability of the data to the network, making it available to miners. 
+A miner accepting a deal pulls the data from the client (or any other host) using the IPFS block service [bitswap protocol](https://github.com/ipfs/specs/tree/master/bitswap).
+
+A miner packs pieces from many clients into a sector, which is then sealed with a proof (aka commitment). 
+Sealing is a computationally intensive process that can take tens of minutes. 
+A client polls the miner directly for deal status to monitor the piece being received, staged in a sector, the sector sealed, and the proof posted to the blockchain. 
+Once the sector commitment is on chain, the client can observe it directly. 
+A miner periodically computes proofs for all sealed sectors and posts on chain. 
+There is no on-chain mapping of pieces to sectors; a client must keep track of its own pieces.
+
+Note that the mechanisms for communication of deals and state are not specified in the protocol, except the format of messages and the eventual on-chain commitment. 
+Other mechanisms may be used.
+
+The storage [client commands](https://github.com/filecoin-project/go-filecoin/blob/master/commands/client.go) interface to a `go-filecoin` daemon in the same way as other node [commands](#commands). 
+Right now, a client must be running a full node, but that’s not in-principle necessary. 
+Future reorganisation will allow the extraction of a thin client binary. 
+
+Data preparation is entirely the responsibility of the client, including breaking it up into appropriate chunks (<= sector size), compressing, 
+encrypting, adding error-correcting codes, and replicating widely enough to achieve redundancy goals. 
+In the future, we will build a client library which handles many of these tasks for a user. 
+We also plan support for “repair miners”, to whom responsibility can be delegated for monitoring and repairing faults.
+
+### Retrieval
+Retrieval mining is not necessarily linked to storage mining, although in practise we expect all storage miners to also run retrieval miners. 
+Retrieval miners serve requests to fetch content, and are not much more than a big cache and some logic to find missing pieces.
+
+The retrieval protocol and implementation are not yet fully developed. 
+At present, retrieval is not charged for, and always causes unsealing of a sector.
+
+Note that due to use of the IPFS block service for transmitting pieces, a client can issue a blockservice `get` which will find pieces in the block service of the local or any other node, 
+and fetch them without invoking retrieval protocol. 
+With time, the block service will evict stale blocks, so this is not reliable. 
```

Comment:
@whyrusleeping here too

Reply:
this is a side effect and should not be relied on. I hesitate to even include it in any documentation.

Reply:
Good call, removed.